src/main/java/com/videoplaza/dataflow/pubsub/source/task/convert/SourceRecordFactory.java

```diff
@@ -0,0 +1,34 @@
+package com.videoplaza.dataflow.pubsub.source.task.convert;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.source.SourceRecord;
+
+import static java.util.Collections.singletonMap;
+import static java.util.Objects.requireNonNull;
+
+public class SourceRecordFactory {
+
+   private final String subscription;
+   private final String topic;
+
+   public SourceRecordFactory(String subscription, String topic) {
+      this.subscription = requireNonNull(subscription, "Subscription is required");
+      this.topic = requireNonNull(topic, "Topic is required");
+   }
+
+   //TODO pass extra message attributes as headers
+   //TODO make key type/schema configurable
```

Comment:
Should these TODO's be resolved in the scope of the PR? If not, maybe create issues to track them.